### DISCUSSION TITLE
Are PTZ cameras supported?

### DISCUSSION THREAD

**Discussion by eliluong** [blakeblackshear/frigate#3684] [Ask A Question] [answered]
Hi all, I wanted to get started with Frigate and a generic camera for some learning. In the documentation, it lists the 5442-series cameras as being a good choice. I found something similar (https://www.amazon.com/EmpireTech-IPC-T5442T-ZE-Vari-Focal-Eyeball-Starlight/dp/B08C77TNY9), which has PTZ. Is the PTZ function used in Frigate? And as an aside, are PTZ generally useful? After looking at Frigate documentation, I didn't see any mention of PTZ. I don't want to waste money on an unnecessary feature. Thank you.

**Comment by blakeblackshear** [comment, marked as answer]
No aspect of PTZ is currently supported in frigate. It may be supported in the future.

**Comment by JakobTewes** [comment]
Oh....pretty interested in PTZ cam support too. Can we have a feature request on this plz?

**Reply by NickM-27** [reply]
There's already a FR https://github.com/blakeblackshear/frigate/issues/1179

**Comment by thinkloop** [comment]
> are PTZ generally useful?

For me personally the only use is to more easily setup the initial angle during installation if the camera is very high-up or hard-to-reach. Otherwise, I have had zero use for my ptz features. Note also that frigate is web-based which inherently comes with 5s-10s lag given the limitations of the web framework, so ptz will be even less useful in this case. Finally, even if everything is perfect, the ptz itself is never what you think. There is a lag in sending messages to the camera, then the motor actuator has a lag, so the experience ends up jumpy and awkward. If you're looking for a Minority Report experience, 2022 ptz is not it (from what I have had experience with).

**Reply by eliluong** [reply]
I imagine the camera or NVR software to spot something of interest then track and zoom. Something like https://youtu.be/akCXeDJxtoE?t=121 which has the motion tracking algorithm built-in, but even that seems to have issues.

**Comment by nibagit** [comment]
Just being able to zoom or check stuff out remotely...always have camera return to standard view.  During manual control any recording, snapshots, motion, etc... should switched off to not end up with odd or unwanted footage

**Comment by skstussy** [comment]
I had it was working in my Home Assistant Frigate installation for a while. It was my go-to when showing off the home automation.  However after several Frigate and other system updates it broke. +-4 months back. First the scripts stopped executing when clicking the PTZ controls. then the PTZ controls disappeared and no longer display. very unhappy I've lost this capability. The code is unchanged and is still present in my config but no longer works. Would be great if these PTZ controls came back even if unsupported! I suspect one of the updates removed them. I never got a chance to figure out why the rest commands stopped executing before the PTZ controls disappeared. 

To display the PTZ controls I added below to the Frigate card.

      elements:
          - type: custom:frigate-card-ptz
            orientation: horizontal
            style:
              transform: none
              left: 20px
              top: 75%
            service: script.camera_ptz
            data_left:
              direction: left
            data_right:
              direction: right
            data_up:
              direction: up
            data_down:
              direction: down
            data_zoom_in:
              direction: zoom_in
            data_zoom_out:
              direction: zoom_out

I had created some scripts to execute when buttons pressed and added separate buttons below the video which set which camera to use as the base url:

  - type: horizontal-stack
    cards:
      - show_name: true
        show_icon: false
        type: button
        tap_action:
          action: call-service
          service: script.set_ptz_url_entrance
        entity: script.set_camera_url_entrance
        name: PTZ Entrance
        show_state: false
        icon: mdi:arrow-all
      - show_name: true
        show_icon: false
        type: button
        tap_action:
          action: call-service
          service: script.set_ptz_url_frontyard
        entity: script.set_camera_url_frontyard
        name: PTZ Front Yard
        show_state: false
      - show_name: true
        show_icon: false
        type: button
        tap_action:
          action: call-service
          service: script.set_ptz_url_backyard
        entity: script.set_camera_url_backyard
        name: PTZ Back Yard
        show_state: false


config scripts:


#PTZ Camera configuration
#This is working as a variable. need to figure out how to get current camera entity to trigger update
variable:
  camera_pt_url:
    value: ''
    attributes:
      previous: ''
      icon: mdi:arrow-all
    restore: true

  camera_z_url:
    value: ''
    attributes:
      previous: ''
      icon: mdi:arrow-all
    restore: true

#### Rest Commands ####
## https://www.home-assistant.io/integrations/rest_command/ ##
rest_command:

##Camera PTZ##
  #Pan and Tilt
  cam_pt_momentary:
    url: "{{ states('variable.camera_pt_url') }}"
    method: PUT
    content_type: "application/xml"
    payload: '<PTZData>
      <pan>{{ pan }}</pan>
      <tilt>{{ tilt }}</tilt>
      <Momentary>
        <duration>50</duration>
      </Momentary>
    </PTZData>'
    username: !secret cctv_username
    password: !secret cctv_password
    verify_ssl: false
  #Zoom
  cam_z_momentary:
    url: "{{ states('variable.camera_z_url') }}"
    method: PUT
    content_type: "application/xml"
    payload: '<PTZData>
      <zoom>{{ zoom }}</zoom>
    </PTZData>'
    username: !secret cctv_username
    password: !secret cctv_password
    verify_ssl: false

**Reply by skstussy** [reply]
Actually the PTZ controls are showing now. I'll try debug why it's not working.
